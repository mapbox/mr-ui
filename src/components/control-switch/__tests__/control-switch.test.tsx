import React from 'react';
import ControlSwitch from '../control-switch';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('ControlSwitch', () => {
  let wrapper;

  describe('basic', () => {
    const props = {
      id: 'test',
      label: 'Basic',
      onChange: jest.fn()
    };

    beforeEach(() => {
      wrapper = render(<ControlSwitch {...props} />)
    });

    test('renders', () => {
      expect(wrapper.baseElement).toMatchSnapshot();
    });

    test('onChange gets called with input id and value', async () => {
      await userEvent.click(wrapper.getByTestId('test-switch'));
      expect(props.onChange).toHaveBeenCalledTimes(1);
      expect(props.onChange).toHaveBeenCalledWith(true, 'test');
    });
  });

  describe('disabled', () => {
    test('disabled state works', async () => {
      const props = {
        id: 'test',
        label: 'Basic',
        onChange: jest.fn(),
        disabled: true
      };

      const { getByTestId } = render(<ControlSwitch {...props} />)
      const switchControl = getByTestId('test-switch');
      await userEvent.click(switchControl);
      expect(props.onChange).not.toHaveBeenCalled();
      expect(switchControl).toHaveProperty('disabled', true);
    });
  });
});
