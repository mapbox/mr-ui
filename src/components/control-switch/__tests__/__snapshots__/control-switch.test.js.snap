// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlSwitch all options renders 1`] = `
<ControlWrapper
  id="testinput"
  themeControlWrapper="bg-red-light"
  validationError="oh no!"
>
  <label
    className="switch-container txt-s"
  >
    <input
      aria-invalid={true}
      autoFocus={true}
      checked={true}
      data-test="testinput-input"
      id="testinput"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="switch switch--red switch--dot-yellow"
    />
    <span
      className="color-white txt-m"
    >
      <span>
        All options
      </span>
      <span
        className="ml3 txt-normal"
      >
        (optional)
      </span>
    </span>
  </label>
</ControlWrapper>
`;

exports[`ControlSwitch basic onChange gets called with input id and value 1`] = `
<ControlWrapper
  id="testinput"
  themeControlWrapper=""
>
  <label
    className="switch-container "
  >
    <input
      checked={false}
      data-test="testinput-input"
      id="testinput"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="switch switch--s-label"
    />
    <span
      className="txt-s ml6"
    >
      Basic
    </span>
  </label>
</ControlWrapper>
`;

exports[`ControlSwitch basic renders 1`] = `
<ControlWrapper
  id="testinput"
  themeControlWrapper=""
>
  <label
    className="switch-container "
  >
    <input
      checked={false}
      data-test="testinput-input"
      id="testinput"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="switch switch--s-label"
    />
    <span
      className="txt-s ml6"
    >
      Basic
    </span>
  </label>
</ControlWrapper>
`;

exports[`ControlSwitch disabled renders 1`] = `
<ControlWrapper
  id="testinput"
  themeControlWrapper=""
>
  <label
    className="switch-container "
  >
    <input
      checked={false}
      data-test="testinput-input"
      disabled={true}
      id="testinput"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="switch switch--s-label"
    />
    <span
      className="txt-s ml6"
    >
      Disabled
    </span>
  </label>
</ControlWrapper>
`;
